byron/semantics/executable-spec/src/Control/State/Transition.hs

```diff
@@ -80,6 +80,11 @@ class ( Eq (PredicateFailure a)
   -- to fail.
   data PredicateFailure a :: *
 
+  -- | Predicate to compare equality of `PredicateFailure`s. Used for
+  -- `traceWithProfile`-related functions.
+  eqPF :: PredicateFailure a -> PredicateFailure a -> Bool
```

Comment:
Agreed that it doesn't belong in the `STS` typeclass. Thinking more about it, though, this can be lifted out into a separate typeclass in `goblins-sts-breeder` (the package I have for breeding genomes). We no longer depend on `eqPF` for matching `PredicateFailure`s in the signal generators, because we no longer pass in a `PredicateFailure` but pass in the valid-or-invalid generator.

So this is only needed in the breeder code, and can be yanked from `c-s-l` to reduce clutter.

Reply:
I've lifted this out into a separate typeclass in the breeder package: https://github.com/input-output-hk/goblins-sts-breeder/commit/f964d1750c78711f7b23786f745ab4f02fba050f#diff-acf82986944a1ad97a87e8f50d4b2b80R24